source/clustering.md

```diff
@@ -678,95 +654,60 @@ The data colored by the cluster assignments returned by K-means.
 ```{index} see: WSSD; K-means inertia
 ```
 
-As mentioned above, we also need to select K by finding
-where the "elbow" occurs in the plot of total WSSD versus the number of clusters.
-We can obtain the total WSSD (inertia) from our
-clustering using `.inertia_` function. For example:
+As mentioned above,
+instead of arbitrarily setting K to a number as we did above,
+we can find the best value for K
+by finding where the "elbow" occurs in the plot of total WSSD versus the number of clusters.
+The total WSSD is stored in the `.inertia_` attribute
+of the clustering object ("inertia" is another term for WSSD).
 
 ```{code-cell} ipython3
 penguin_clust.inertia_
 ```
 
 To calculate the total WSSD for a variety of Ks, we will
-create a data frame with a column named `k` with rows containing
-each value of K we want to run K-means with (here, 1 to 9).
+create a data frame that contains different values of `k`
+and the WSSD of running KMeans with each values of k.
+To create this dataframe,
+we will use what is called a "for loop"
+where we repeat an operation a number of times.
+Here is an examples of a for loop that simply prints each number from 1-9:
 
 ```{code-cell} ipython3
-import numpy as np
-penguin_clust_ks = pd.DataFrame({"k": np.array(range(1, 10)).transpose()})
+numbers = range(1, 10)
+for number in numbers:
+    print(number)
 ```
 
-```{index} pandas.DataFrame; assign
-```
-
-Then we use `assign()` to create a new column and `lambda` operator to apply the `KMeans` function
-within each row to each K.
+Next,
+we will use this approach to compute the WSSD/inertia for the K-values 1 through 9,
+and temporarily store these values in a list
+that we will use to create a dataframe of both the K-values and their corresponding WSSDs/inertias.
 
 ```{code-cell} ipython3
-np.random.seed(12)
-penguin_clust_ks = penguin_clust_ks.assign(
-    penguin_clusts=penguin_clust_ks['k'].apply(
-        lambda x: KMeans(n_clusters=x, n_init=3, init="random").fit(standardized_data)
-    )
-)
-```
+# Create an empty list
+inertias = []
 
-If we take a look at our data frame `penguin_clust_ks` now,
-we see that it has two columns: one with the value for K,
-and the other holding the clustering model objects.
+ks = range(1, 10)
+for k in ks:
+    # Save the computed inertia for each k
+    inertias.append(KMeans(n_clusters=k).fit(standardized_data).inertia_)
```

Comment:
I think this is more straight-forward than the current approach with apply. Since we already are teaching for loops in chapter 10 I thought it made sense to have one here too. We can of course discuss removing them at both places if we don't think we need them.